### PR TITLE
Avoid specific version pinning inside setup.py

### PR DESCRIPTION
**Pull request by chapmanb**
The current cwltool fails with pathlib 2.3.3 (the latest) installed:

```
pkg_resources.ContextualVersionConflict: (pathlib2 2.3.3 (lib/python2.7/site-packages), Requirement.parse('pathlib2==2.3.2; python_version < "3"'), set(['cwltool']))
```
due to very specific version pinning in cwltool's setup.py:

https://github.com/common-workflow-language/cwltool/commit/0c6a7033d393a42bb6598644422f1e0fa75c1186

Rather than using specific pins in the recipe, which is typically
difficult to maintain, this relaxes the very specific checks in
setup.py.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ requirements:
   run:
     - bagit >=1.6.4
     - mypy_extensions
-    - prov ==1.5.1
```
Comment (croth1) [inline recipe/meta.yaml]: Unfortunately we cannot do this due to https://github.com/common-workflow-language/cwltool/issues/1000. Instead I suggest we try to rebuild prov-1.5.1 for noarch here: https://github.com/conda-forge/prov-feedstock/issues/10.

**Comment by croth1** [comment]
@chapmanb, I just saw that this PR was opened by accident from a branch pushed to the CF repository instead of from a fork. This led to the package already being built and uploaded to anaconda.org. I will close the branch and ask core to remove the uploaded packages.

**Comment by chapmanb** [comment]
Christian;
Yikes, sorry. That is my mistake. My conda-forge skills are pretty weak and I didn't realize I was out of sync with the right approach. I'll make sure to fork going forward. Thanks for cleaning up my mess.

**Comment by croth1** [comment]
@chapmanb, no worries. Once core has the chance to delete the other builds, we will merge #3 and everything is back to normal 👍

**Pull request closed without merging**